R/provider-google.R

```diff
@@ -105,11 +106,28 @@ chat_google_vertex <- function(
 }
 
 # https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.endpoints/generateContent
-vertex_url <- function(location, project_id) {
-  paste_c(
-    c("https://", location, "-aiplatform.googleapis.com/v1"),
+vertex_url <- function(location, project_id, list_models = FALSE) {
```

Comment:
I think this has become too convoluted in trying to support the model listing endpoint. You could move that logic into the `models_google_vertex()` function below instead.